spec/services/data_migrations/backfill_withdrawal_reasons_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'rails_helper'
+
+RSpec.describe DataMigrations::BackfillWithdrawalReasons do
+  context 'where choice has multiple valid withdrawal reasons' do
+    it 'maps the old reasons to the new reasons' do
+      application_choice = create(:application_choice, :withdrawn, structured_withdrawal_reasons: all_old_reasons)
+      described_class.new.change
+      expect(application_choice.reload.structured_withdrawal_reasons).to eq(all_new_reasons)
+    end
+
+    it 'does not change the timestamps' do
+      updated_at = Time.zone.local(2023, 12, 12, 9)
+      application_choice = create(:application_choice,
+                                  :withdrawn,
+                                  structured_withdrawal_reasons: all_old_reasons,
+                                  updated_at: updated_at)
+      described_class.new.change
+      expect(application_choice.reload.updated_at).to be_within(0.1.seconds).of(updated_at)
+    end
+  end
+
+  context 'where choice has some invalid withdrawal reasons' do
+    it 'discards to the invalid reason, and maps the valid ones to the new choices' do
+      application_choice = create(:application_choice, :withdrawn, structured_withdrawal_reasons: %w[costs something_else])
+      described_class.new.change
+      expect(application_choice.reload.structured_withdrawal_reasons).to eq(['concerns_about_cost'])
+    end
+  end
+
+  context 'where choice does not have any withdrawal reasons' do
+    it 'does not change the structured withdrawal reasons' do
+      application_choice = create(:application_choice, :withdrawn, structured_withdrawal_reasons: [])
```

Comment:
Yes
- `Array.wrap("") => []`
- `Array.wrap(nil) => [nil]`
- `Array.wrap('something') => ['something']`
- `Array.wrap(['something']) => ['something]`

But this is actually being paranoid because
- I've checked and we don't have any records where the value is nil. 
- AND you can try this on the command line -- try to save a `structured_withdrawal_reasons` as "" or "costs". It will coerce it to an empty array.

Reply:
Awesome - I read the `Arrap.wrap` and though "yass we're using Array.wrap to handle all the weird things" and then I forgot about it when I finished reviewing 🤦‍♂️